internal/saucecloud/cloud.go

```diff
@@ -103,8 +103,12 @@ func (r *CloudRunner) collectResults(artifactCfg config.ArtifactDownload, result
 	inProgress := expected
 	passed := true
 
-	junitRequired := report.IsArtifactRequired(r.Reporters, report.JUnitArtifact)
-	jsonResultRequired := report.IsArtifactRequired(r.Reporters, report.JSONArtifact)
+	var junitRequired bool
+	var jsonResultRequired bool
+	if !r.Async {
+		junitRequired = report.IsArtifactRequired(r.Reporters, report.JUnitArtifact)
+		jsonResultRequired = report.IsArtifactRequired(r.Reporters, report.JSONArtifact)
+	}
```

Comment:
Hm. Is this location better than https://github.com/saucelabs/saucectl/blob/cbea780dfda049c6f782b322860d93d86e4f7798/internal/cmd/run/run.go#L295-L306 ?

My thinking: Why init a reporter in the first place if it's not going to be used?

Reply:
Refine the implementation. I moved `report.IsArtifactRequired(r.Reporters, report.JUnitArtifact)` inside of the expecting job loop. The implementation is clearer, although it slightly affects performance.